vars/promoteArtifacts.groovy

```diff
@@ -29,55 +28,57 @@ void call(Map args = [:]) {
 
     print("Actions ${fileActions}")
 
+    argsMap = [:]
+    argsMap['signatureType'] = '.sig'
+
+    //////////// Signing Artifacts
+    println("Signing Core Pluings")
+    String corePluginDir = "$WORKSPACE/artifacts/$artifactPath/builds/$filename/core-plugins"
+    argsMap['artifactPath'] = corePluginDir
+    for (Closure action : fileActions) {
+        action(argsMap)
+    }
+
+    println("Signing TAR Artifacts")
+    String coreFullPath = ['core', filename, version].join('/')
+    String bundleFullPath = ['bundle', filename, version].join('/')
+    for (Closure action : fileActions) {
+        for (file in findFiles(glob: "**/${filename}-min-${version}*.tar.gz,**/${filename}-${version}*.tar.gz")) {
+            argsMap['artifactPath'] = "$WORKSPACE" + "/" + file.getPath()
+            action(argsMap)
+        }
+    }
+
+
+
+
+    //////////// Uploading Artifacts
     withAWS(role: "${ARTIFACT_PROMOTION_ROLE_NAME}", roleAccount: "${AWS_ACCOUNT_ARTIFACT}", duration: 900, roleSessionName: 'jenkins-session') {
-        // Core Plugins
-        println("Start Core Plugin Promotion to artifects.opensearch.org Bucket")
         List<String> corePluginList = buildManifest.components.artifacts."core-plugins"[0]
         for (String pluginSubPath : corePluginList) {
             String pluginSubFolder = pluginSubPath.split('/')[0]
```

Comment:
This (and possibly other assignments below) are no longer used, remove.

OpenSearch Dashboards doesn't have `core-plugins`, what's going to happen here?

Reply:
@peterzhuamazon

Reply:
We will need to add a condition to push core-plugins for OpenSearch only, the rest are for both the opensearch/dashboards.

Reply:
This tells me that `core-plugins` is one of the things to publish, and needs to come in as an argument instead of being hard-coded. Provide the list of the components to publish from the caller?